lib/medtronic/medtronicSimulator.js

```diff
@@ -66,6 +77,111 @@ exports.make = function(config){
   }
 
   return {
+    basal: function(event) {
+      ensureTimestamp(event);
+
+      if (currBasal != null) {
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
+        }
+
+        if(currBasal.deliveryType === 'temp' && currPumpSettings) {
+          var currSchedule = currPumpSettings.basalSchedules[currPumpSettings.activeSchedule];
+          var startTime = common.computeMillisInCurrentDay(currBasal);
+          var endTime = startTime + currBasal.duration;
+
+          for(var i in currSchedule) {
+            if((startTime < currSchedule[i].start) && (endTime > currSchedule[i].start)) {
+              // there was a schedule change _during_ the temp basal
+              var adjustedDuration = currSchedule[i].start - startTime;
+              var oldDuration = currBasal.duration;
+              currBasal.with_duration(adjustedDuration)
+                       .with_expectedDuration(oldDuration);
```

Comment:
IMO, this is not a case for the use of `expectedDuration`, and I don't believe we've used it as such for schedule changes during temps in the old CareLink parser. I also just checked in my Tandem data because I conveniently had a schedule-crossing basal running last night, and the first segment of the temp-that-crosses-boundaries does not have an `expectedDuration` there. (Now, perhaps we don't have that info for Tandem, can't remember, but still I think `expectedDuration` does not exactly belong in this case.) However, it does beg the question of how to represent it when the temp basal as a whole is cut short. Say I program a temp basal lasting 6 hours. It crosses a schedule boundary after 1.25 hours and gets split into two segments, then I cancel it four hours in. I don't think we have a way to represent the original "collective" `expectedDuration` in that case with our current data model...